data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/configuration/KinesisStreamPollingConfig.java

```diff
@@ -0,0 +1,17 @@
+package org.opensearch.dataprepper.plugins.kinesis.source.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+
+public class KinesisStreamPollingConfig {
+    private static final int DEFAULT_MAX_RECORDS = 10000;
+    private static final int IDLE_TIME_BETWEEN_READS_IN_MILLIS = 250;
+    @Getter
+    @JsonProperty("maxPollingRecords")
+    private int maxPollingRecords = DEFAULT_MAX_RECORDS;
+
+    @Getter
+    @JsonProperty("idleTimeBetweenReadsInMillis")
+    private int idleTimeBetweenReadsInMillis = IDLE_TIME_BETWEEN_READS_IN_MILLIS;
```

Comment:
```suggestion
    private Duration idleTimeBetweenReads = IDLE_TIME_BETWEEN_READS;
```

Use `Duration`. You would also need to change `IDLE_TIME_BETWEEN_READS` to a Duration.

Reply:
Thanks @dlvenable. I have fixed this.